File: 2024/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSub = nums[0]
        curSome = 0

        for n in nums:
            if curSome < 0:
                curSome = 0
            curSome += n
            maxSub = max(curSome, maxSub)
        return maxSub

File: 2024-mk2/dp/maximum-subarray.go
func maxSubArray(nums []int) int {
    curMax := nums[0]
    max := nums[0]

    for i:=1; i<len(nums); i++ {
        if curMax+nums[i] > nums[i] {
            curMax = curMax+nums[i]
        } else {
            curMax = nums[i]
        }
        if curMax > max {
            max = curMax
        }
    }
    return max
}
